### ISSUE TITLE
Architecture diagram?

### ISSUE THREAD

**Issue by brylie** [clydeio/clydeio#22]
As a developer
I would like a diagram of the project architecture (including planned components)
so that I can quickly and visually comprehend the project design

Is there any diagram, even a simple drawing, depicting the initial design plans for this project? I have found the diagrams in the [I'm working on Clyde IO](http://www.acuriousanimal.com/2015/09/14/i-m-working-on-clydeio.html) blog post useful, but lacking in architectural detail.

![Clyde IO](http://acanimal.github.io/assets/uploads/2015-09-clyde.png)
![Clyde IO filters](http://acanimal.github.io/assets/uploads/2015-09-clyde-dataflow.png)

**Comment by acanimal** [comment]
What kind of diagram do you need? Which things do you need to be reflected?

On its base clyde is simply a app that read a configuration and connects a set of middlewares in the specified order. It is up to you to decide which middlewares (filters) execute for each request.

**Comment by brylie** [comment]
Perhaps a basic diagram listing all the different filters, base compobent(s), and other relevant components such as routing, storage adapters, etc.

On 17 September 2015 17:02:09 EEST, acanimal notifications@github.com wrote:

> What kind of diagram do you need? Which things do you need to be
> reflected?
> 
> On its base clyde is simply a app that read a configuration and
> connects a set of middlewares in the specified order. It is up to you
> to decide which middlewares (filters) execute for each request.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/clydeio/clydeio/issues/22#issuecomment-141095990

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by acanimal** [comment]
Yes, I need to improve much more the documentation. I will do once finished the configuration module.

As a summary the filters (middlewares) contains all the functionalities: cache, rate limiting, authentication, etc, Clyde core is only the engine that runs them in the right order.
